cdap-data-fabric/src/main/java/io/cdap/cdap/spi/data/nosql/NoSqlStructuredTable.java

```diff
@@ -479,17 +566,70 @@ protected StructuredRow computeNext() {
 
     @Override
     public void close() {
-      closeScanner();
+      currentScanner.close();
     }
 
     private void closeScanner() {
       if (currentScanner != null) {
-        currentScanner.close();
```

Comment:
still should be closing it here, otherwise the iterator will never be closed.

Reply:
yes, that was removed accidentally. Thanks!